README.md

```diff
@@ -17,3 +17,379 @@ Implement a prototype job worker service that provides an API to run arbitrary L
 
 * Client should be able to query result of the job execution and fetch the logs
 
+## Design Document
+
+### Limitations And Scope
+
+#### Data Management
+Though the use of a database to store persistant data would be ideal, I will be instead storing the outputs and error outputs into logs stored on 
+the file system of the linux worker. The logs will be generated at start time with the foldername \<uuid\>-\<startTimeStamp\>
+The client can query the server to see what is running to determine what can be killed and a query a list of jobs that were executed,
+or the client can store response data from server. I'll be using a self-implemented Data Store to store in memory data to manage jobs ran by server.
+
+#### Scale
+The scope of this project would only deal with a single linux worker server interfacing with multiple clients
+
+### API Design
+
+* The client will have an Command Api that takes in three types of request, Start, Stop, and Query
+
+* The client can then execute the commands over the server via Execute APIs'
+
+#### Start
+
+* The Start command is called with a StartRequest that has the client's command and required arguments and optional env, dir params
+
+* A uuid (universal unique identification) will be generated and a folder called START-\<startTimeStamp\>will be created, two logs called stdout.log and stderr.log will be created
+
+* the start command will execute the job and return with the uuid, pid, startTimeStamp, if it fails the process table will be correspondingly updated
+
+* goroutines should manage running processes in the background (outputing into logs, updating dataStore)
+
+* when the job is done the process table will be updated
+
+```
+type StartRequest {
+    //path of a program or executable
+    string command
+    //arguments to invoke program or executable with
+    []string args 
+    // environment variables for the execution
+    //OPTIONAL variable
+    []string env
+    // current working directory of the execution
+    //OPTIONAL variable, default will be the working directory of server 
+    string dir 
+}
+
+type StartResponse {
+    //process Id of the running process executed by the command
+    //will be 0 if process failed to execute
+    int pid 
+    //univeral unique identifier that tags each unique request made to server
+    string uuid
+    //starting time of start request
+    string startTimeStamp
+    string status
+    int statusCode
```

Comment:
what are the possible status codes?

Reply:
my original intention was handling the errors via grpc codes, so if the response failed somehow it would be reflected in grpc error format, would it be okay to remove the status and statusCode fields from the response types then